awswrangler/_utils.py

```diff
@@ -414,7 +415,7 @@ def table_refs_to_df(
 ) -> pd.DataFrame:
     """Build Pandas dataframe from list of PyArrow tables."""
     if isinstance(tables[0], pa.Table):
-        return ensure_df_is_mutable(pa.concat_tables(tables, promote=True).to_pandas(**kwargs))
+        return _table_to_df(pa.concat_tables(tables, promote=True), kwargs=kwargs)
```

Comment:
This is an example of using `_table_to_df` defined in the `arrow` module in order to ensure this conversion is consistent across the codebase